indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/autoscaler/LagBasedAutoScaler.java

```diff
@@ -155,13 +154,7 @@ private Runnable computeAndCollectLag()
       LOCK.lock();
       try {
         if (!spec.isSuspended()) {
-          LagStats lagStats = supervisor.computeLagStats();
-          if (lagStats == null) {
-            lagMetricsQueue.offer(0L);
-          } else {
-            long lag = lagStats.get(supervisor.getLagMetricForAutoScaler());
-            lagMetricsQueue.offer(lag > 0 ? lag : 0L);
-          }
+          lagMetricsQueue.offer(supervisor.computeLagForAutoScaler());
```

Comment:
I think we still need to handle the case where the returned lag is negative.

```suggestion
          long lag = supervisor.computeLagForAutoScaler();
          lagMetricsQueue.offer(lag > 0 ? lag : 0L);
```